v1/dns/index.ts

```diff
@@ -1 +1,50 @@
-// STUB
+import dns2 from 'dns2'
+import { FastifyBaseLogger } from 'fastify'
+import { SiteConfigStore } from '../config/sites.js'
+
+export async function initDnsServer (port: number, store: SiteConfigStore, logger: FastifyBaseLogger): Promise<ReturnType<typeof dns2.createServer>> {
+  const server = dns2.createServer({
+    udp: true,
+    handle: (request, send, rinfo) => {
+      const response = dns2.Packet.createResponseFromRequest(request)
+      const [{ name }] = request.questions
+      logger.info(`[dns] ${rinfo.address}:${rinfo.port} asked for ${name}`)
+
+      const trimmedName = name.replace('_dnslink.', '')
+      store.get(trimmedName)
+        .then(({ links }) => {
+          // TODO(support http and hyper)
+          if (links.ipfs !== undefined) {
+            response.answers.push({
+              name,
+              type: dns2.Packet.TYPE.TXT,
+              class: dns2.Packet.CLASS.IN,
+              ttl: 60,
+              data: `dnslink=${links.ipfs.link}`
```

Comment:
The link should be converted from `ipns://KEY` to  `/ipns/KEY`

Note that instead of using the `link` we should be using the `pubKey` URL

Similarly we should use `dnslink=/hyper/'` for hyper, and using it's 

Make sure the `hyper` one comes after IPFS dnslink

Reply:
I think you may have pressed enter too early for your comment on Hyper haha! Can you clarify what the link structure for hyper looks like? Is it sufficient to just do `dnslink=/hyper/${links.hyper.link}`

Reply:
Can you clarify what you meant by 

> One other change that should be made, the link for IPFS/Hyper should use the DNS domain instead of the public key so that DP users will be linked to the "nice" URL

I thought that `link` would have been the nice URL but perhaps I don't understand how the URLs map in IPFS

Reply:
> an you clarify what the link structure for hyper looks like? Is it sufficient to just do dnslink=/hyper/${links.hyper.link}

Whoops, forgot to finish my sentance there. It should be `dnslink=/hyper/${links.hyper.raw.slice('hyper://').length)}` Note how we want the raw public key and not the DNS domain which should be placed in the `link`.

Reply:
> I thought that link would have been the nice URL but perhaps I don't understand how the URLs map in IPFS

At the moment DP produces links that look like `ipns://IPNS_KEY_HERE` like `ipns://bafywhateverelsehere`. This is the "raw" link for the public key and is not human readable.

The "nice" URL is one that contains the domain name (which is what this DNS server resolves) like `ipns://distributed.press`.

The DNS server needs to serve TXT entries which link to the raw public keys of the sites since that's how clients turn the DNS domain name into the p2p URL that they can then resolve to the actual data.